prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -412,6 +422,132 @@ func Run(o *Options) error {
+
+			if !changed {
+				// TODO(chaodaiG): decide how to handle this with multiple commits
+				logrus.Info("Nothing changed, exiting ...")
+				return nil
+			}
+
+			if err := gitCommit(o.GitName, o.GitEmail, msg, stdout, stderr, false); err != nil {
+				return fmt.Errorf("failed git commit: %w", err)
+			}
+		}
+
+		if err := gitPush(fmt.Sprintf("git@github.com:%s/%s.git", o.GitHubLogin, o.RemoteName),
+			o.HeadBranchName, stdout, stderr, o.SkipPullRequest); err != nil {
+			return fmt.Errorf("failed to push changes to the remote branch: %w", err)
+		}
+
+		summary, body, err := prh.PRTitleBody()
+		if err != nil {
+			return fmt.Errorf("failed creating PR summary and body: %w", err)
+		}
+		if o.GitHubBaseBranch == "" {
+			repo, err := gc.GetRepo(o.GitHubOrg, o.GitHubRepo)
+			if err != nil {
+				return fmt.Errorf("failed to detect default remote branch for %s/%s: %w", o.GitHubOrg, o.GitHubRepo, err)
+			}
+			o.GitHubBaseBranch = repo.DefaultBranch
+		}
+		if err := updatePRWithLabels2(gc, o.GitHubOrg, o.GitHubRepo,
+			getAssignment(o.AssignTo, o.OncallAddress, o.OncallGroup), o.GitHubLogin,
+			o.GitHubBaseBranch, o.HeadBranchName, updater.PreventMods, summary, body, o.Labels, o.SkipPullRequest); err != nil {
+			return fmt.Errorf("failed to create the PR: %w", err)
+		}
+	} else {
+		if err := Call(stdout, stderr, gitCmd, "config", "http.cookiefile", o.Gerrit.CookieFile); err != nil {
+			return fmt.Errorf("unable to load cookiefile: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "config", "user.name", o.Gerrit.Author); err != nil {
+			return fmt.Errorf("unable to set username: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "config", "user.email", o.Gerrit.Email); err != nil {
+			return fmt.Errorf("unable to set password: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "remote", "add", "upstream", o.Gerrit.HostRepo); err != nil {
+			return fmt.Errorf("unable to add upstream remote: %v", err)
+		}
+		changeId, err := getChangeId(o.Gerrit.Author, o.Gerrit.AutobumpPRIdentifier, "")
+		if err != nil {
+			return fmt.Errorf("Failed to create CR: %w", err)
+		}
+
+		// Make change, commit and push
+		for _, f := range prh.Changes() {
+			msg, err := f()
+			if err != nil {
+				return fmt.Errorf("failed to process provided function: %w", err)
+			}
+
+			changed, err := HasChanges()
+			if err != nil {
+				return fmt.Errorf("error occurred when checking changes: %w", err)
+			}
+
+			if !changed {
+				// TODO(chaodaiG): decide how to handle this with multiple commits
+				logrus.Info("Nothing changed, exiting ...")
+				return nil
+			}
+
+			err = gerritCommitandPush2(msg, o.Gerrit.AutobumpPRIdentifier, changeId, nil, nil, stdout, stderr)
+			// If failed to push because a closed PR already exists with this change ID (the PR was abandoned). Hash the ID again and try one more time.
+			if err != nil && strings.Contains(err.Error(), "failed to push some refs") && strings.Contains(err.Error(), "closed") {
```

Comment:
Also a nit, but this would be less error prone if you did string check etc within the `err != nil`, right now the final `if err != nil` is very far away from the error

Reply:
good idea, done